File: medium/2091-removing-minimum-and-maximum-from-array/solution.java
/**
 * 2091. Removing Minimum and Maximum From Array
 * Time: O(n)
 * Space: O(1)
 */
class Solution {
    public int minimumDeletions(int[] nums) {
        int n = nums.length;
        int minIdx = 0, maxIdx = 0;
        for (int i = 1; i < n; i++) {
            if (nums[i] < nums[minIdx]) {
                minIdx = i;
            }
            if (nums[i] > nums[maxIdx]) {
                maxIdx = i;
            }
        }

        int lo = Math.min(minIdx, maxIdx);
        int hi = Math.max(minIdx, maxIdx);

        int bothFront = hi + 1;
        int bothBack = n - lo;
        int split = (lo + 1) + (n - hi);
        return Math.min(bothFront, Math.min(bothBack, split));
    }
}

File: medium/2091-removing-minimum-and-maximum-from-array/solution.py
from typing import List


class Solution:
    def minimumDeletions(self, nums: List[int]) -> int:
        """
        2091. Removing Minimum and Maximum From Array
        Time: O(n)
        Space: O(1)
        """
        n = len(nums)
        min_idx = min(range(n), key=lambda i: nums[i])
        max_idx = max(range(n), key=lambda i: nums[i])
        lo, hi = min(min_idx, max_idx), max(min_idx, max_idx)

        both_front = hi + 1
        both_back = n - lo
        split = (lo + 1) + (n - hi)
        return min(both_front, both_back, split)
